lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,130 @@
+const fs = require('fs-extra');
+const path = require('path');
+const R = require('ramda');
+const async = require('async');
+
+const httpClient = require('@src/clients/http-client');
+const SkillMetadataController = require('@src/controllers/skill-metadata-controller');
+const CliError = require('@src/exceptions/cli-error');
+const ResourcesConfig = require('@src/model/resources-config');
+const Manifest = require('@src/model/manifest');
+const stringUtils = require('@src/utils/string-utils');
+const CONSTANTS = require('@src/utils/constants');
+
+module.exports = {
+    initiateModels,
+    addLocales
+};
+
+/**
+ * Start with loading the project Models global instances.
+ * @param {String} profile
+ */
+function initiateModels(profile) {
+    new ResourcesConfig(path.join(process.cwd(), CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG));
+    const skillPackageSrc = ResourcesConfig.getInstance().getSkillMetaSrc(profile);
+    if (!stringUtils.isNonBlankString(skillPackageSrc)) {
+        throw new CliError('Skill package src is not found in ask-resources.json.');
+    }
+    if (!fs.existsSync(skillPackageSrc)) {
+        throw new CliError(`The skillMetadata src file ${skillPackageSrc} does not exist.`);
+    }
+    const manifestPath = path.join(skillPackageSrc, CONSTANTS.FILE_PATH.SKILL_PACKAGE.MANIFEST);
+    new Manifest(manifestPath);
+}
+
+/**
+ * Add locales by deciding where to get the source of interaction models,
+ * and commit the copy paste from the source to the target location.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @param {Function} callback
+ */
+function addLocales(selectedLocales, profile, doDebug, callback) {
+    const iModelSourceByLocales = _resolveSourcesForIModel(selectedLocales, profile, doDebug);
```

Comment:
_resolveSourcesForIModel was really confusing until I ran in debug mode. may be getNewLocaleSettings?

Reply:
Do you mean we need to make this function as a general pre-processing to get all the settings we need for each new locale?

Reply:
no. just a name was confusing.